### ISSUE TITLE
Move NGEN hyAggregate workflow over?

### ISSUE THREAD

**Issue by mikejohnson51** [dblodgett-usgs/hyRefactor#32]
I would like to move the aggregation workflow from [hyAggregate](https://github.com/mikejohnson51/hyAggregate) over here if you agree.

This will be a large PR so wanted to check first.

**Comment by dblodgett-usgs** [comment]
Yeah -- let's do it. But as we do, I want to make sure we are being diligent about test coverage. 

I need to prioritize #25 and get it merged. Can we plan on you working on a PR after I've had time to do that?

**Comment by mikejohnson51** [comment]
yep! Ill watch that.

**Comment by dblodgett-usgs** [comment]
As we discussed -- there's a chance that our two methods converge around this function:
https://github.com/dblodgett-usgs/hyRefactor/blob/main/R/aggregate_network.R#L77

If, instead of "make_outlets_valid" we were to call a function that returned the required distribution of nexuses (outlets).

From our chat:
> The problem with the distribution approach is that it requires iterative aggregations so by the time the outlets are identified the topology/geometry's are already fully aggregated

Yes -- so don't use https://github.com/dblodgett-usgs/hyRefactor/blob/main/R/aggregate_network.R#L82. 

The key is that you create catchment sets and flowpath sets that get aggregated into a consistent data structure.
